src/helpers/helpers.time.js

```diff
@@ -55,15 +55,28 @@ module.exports = function(Chart) {
 		var ticks = [];
 		if (options.maxTicks) {
 			var stepSize = options.stepSize;
-			ticks.push(options.min !== undefined ? options.min : niceRange.min);
-			var cur = moment(niceRange.min);
-			while (cur.add(stepSize, options.unit).valueOf() < niceRange.max) {
-				ticks.push(cur.valueOf());
+			var startTick = options.min !== undefined ? options.min : niceRange.min;
+			var majorUnitStart = startTick;
+			var majorUnit = options.majorUnit;
+			if (majorUnit) {
+				majorUnitStart = moment(startTick).add(1, majorUnit).startOf(majorUnit);
```

Comment:
you could combine a few lines by doing:

    var majorUnitStart = majorUnit ? moment(startTick).add(1, majorUnit).startOf(majorUnit) : startTick;